test/unit/math/rev/mat/prob/bernoulli_logit_glm_lpmf_test.cpp

```diff
@@ -22,125 +22,205 @@ TEST(ProbDistributionsBernoulliLogitGLM, glm_matches_bernoulli_logit_doubles) {
+    Matrix<double, Dynamic, 1> theta(3, 1);
+    theta = x * beta + alphavec;
+
+    EXPECT_FLOAT_EQ((stan::math::bernoulli_logit_lpmf(y, theta)),
+                    (stan::math::bernoulli_logit_glm_lpmf(y, x, alpha, beta)));
+    EXPECT_FLOAT_EQ(
+        (stan::math::bernoulli_logit_lpmf<true>(y, theta)),
+        (stan::math::bernoulli_logit_glm_lpmf<true>(y, x, alpha, beta)));
+  }
 }
 
 //  We check that the gradients of the new regression match those of one built
 //  from existing primitives.
 TEST(ProbDistributionsBernoulliLogitGLM, glm_matches_bernoulli_logit_vars) {
-  Matrix<int, Dynamic, 1> n(3, 1);
-  n << 1, 0, 1;
+  Matrix<int, Dynamic, 1> y(3, 1);
+  y << 1, 0, 1;
   Matrix<var, Dynamic, Dynamic> x(3, 2);
-  x << -12, 46, -42, 24, 25, 27;
+  x << -1234, 46, -42, 24, 25, 27;
   Matrix<var, Dynamic, 1> beta(2, 1);
-  beta << 0.3, 2;
+  beta << 0.3, 2000;
   var alpha = 0.3;
   Matrix<var, Dynamic, 1> alphavec = alpha * Matrix<double, 3, 1>::Ones();
   Matrix<var, Dynamic, 1> theta(3, 1);
   theta = x * beta + alphavec;
 
-  var lp = stan::math::bernoulli_logit_lpmf(n, theta);
+  var lp = stan::math::bernoulli_logit_lpmf(y, theta);
   lp.grad();
 
+  double lp_val = lp.val();
+  double alpha_adj = alpha.adj();
+  Matrix<double, Dynamic, Dynamic> x_adj(3, 2);
+  Matrix<double, Dynamic, 1> beta_adj(2, 1);
+  for (size_t i = 0; i < 2; i++) {
+    beta_adj[i] = beta[i].adj();
+    for (size_t j = 0; j < 3; j++) {
+      x_adj(j, i) = x(j, i).adj();
+    }
+  }
+
   stan::math::recover_memory();
 
-  Matrix<int, Dynamic, 1> n2(3, 1);
-  n2 << 1, 0, 1;
+  Matrix<int, Dynamic, 1> y2(3, 1);
+  y2 << 1, 0, 1;
   Matrix<var, Dynamic, Dynamic> x2(3, 2);
-  x2 << -12, 46, -42, 24, 25, 27;
+  x2 << -1234, 46, -42, 24, 25, 27;
   Matrix<var, Dynamic, 1> beta2(2, 1);
-  beta2 << 0.3, 2;
+  beta2 << 0.3, 2000;
   var alpha2 = 0.3;
 
-  var lp2 = stan::math::bernoulli_logit_glm_lpmf(n2, x2, beta2, alpha2);
+  var lp2 = stan::math::bernoulli_logit_glm_lpmf(y2, x2, alpha2, beta2);
   lp2.grad();
 
-  EXPECT_FLOAT_EQ(lp.val(), lp2.val());
+  EXPECT_FLOAT_EQ(lp_val, lp2.val());
+  EXPECT_FLOAT_EQ(alpha_adj, alpha2.adj());
   for (size_t i = 0; i < 2; i++) {
-    EXPECT_FLOAT_EQ(beta[i].adj(), beta2[i].adj());
-  }
-  EXPECT_FLOAT_EQ(alpha.adj(), alpha2.adj());
-  for (size_t j = 0; j < 3; j++) {
-    for (size_t i = 0; i < 2; i++) {
-      EXPECT_FLOAT_EQ(x(j, i).adj(), x2(j, i).adj());
+    EXPECT_FLOAT_EQ(beta_adj[i], beta2[i].adj());
+    for (size_t j = 0; j < 3; j++) {
+      EXPECT_FLOAT_EQ(x_adj(j, i), x2(j, i).adj());
     }
   }
 }
 
-//  Here, we compare the speed of the new regression to that of one built from
-//  existing primitives.
+//  We check that the gradients of the new regression match those of one built
+//  from existing primitives.
+TEST(ProbDistributionsBernoulliLogitGLM,
+     glm_matches_bernoulli_logit_vars_rand) {
+  for (size_t ii = 0; ii < 20000; ii++) {
```

Comment:
Repeating this test 20000 times? For this sort of test 5-10 should be find in my opinion. 20k seems like a lot. This applies to all the files.

Reply:
Haha. Yes. This is from when I really didn't trust myself to have done the right thing. This can be reduced now.